File: CS/ProgressTrackBar/MyRepositoryItemTrackBar.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;

namespace WindowsFormsApplication1
{
    class MyRepositoryItemTrackBar : RepositoryItemTrackBar
    {
        ProgressBarControl _progress;
        public ProgressBarControl Progress
        {
            get { return _progress; }
            set
            {
                if (_progress == value)
                    return;
                if (_progress != null)
                {
                    _progress.PropertiesChanged -= new EventHandler(_progress_PropertiesChanged);
                    _progress.EditValueChanged -= new EventHandler(_progress_PropertiesChanged);
                }
                _progress = value;
                if (_progress != null)
                {
                    _progress.PropertiesChanged += new EventHandler(_progress_PropertiesChanged);
                    _progress.EditValueChanged += new EventHandler(_progress_PropertiesChanged);
                }
                this.OnPropertiesChanged(EventArgs.Empty);
            }
        }

        void _progress_PropertiesChanged(object sender, EventArgs e)
        {
            this.OnPropertiesChanged(EventArgs.Empty);
        }

        public MyRepositoryItemTrackBar()
            : base()
        {
        }

        internal const string EditorName = "MyTrackBar";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyTrackBar),
                typeof(MyRepositoryItemTrackBar), typeof(MyTrackBarViewInfo),
                new TrackBarPainter(), true));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }
        public override void Assign(RepositoryItem item)
        {
            base.Assign(item);
            this.Progress = (item as MyRepositoryItemTrackBar).Progress;
        }
    }

}

File: VB/ProgressTrackBar/MyRepositoryItemTrackBar.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing

Namespace WindowsFormsApplication1
	Friend Class MyRepositoryItemTrackBar
		Inherits RepositoryItemTrackBar
		Private _progress As ProgressBarControl
		Public Property Progress() As ProgressBarControl
			Get
				Return _progress
			End Get
			Set(ByVal value As ProgressBarControl)
				If _progress Is value Then
					Return
				End If
				If _progress IsNot Nothing Then
					RemoveHandler _progress.PropertiesChanged, AddressOf _progress_PropertiesChanged
					RemoveHandler _progress.EditValueChanged, AddressOf _progress_PropertiesChanged
				End If
				_progress = value
				If _progress IsNot Nothing Then
					AddHandler _progress.PropertiesChanged, AddressOf _progress_PropertiesChanged
					AddHandler _progress.EditValueChanged, AddressOf _progress_PropertiesChanged
				End If
				Me.OnPropertiesChanged(EventArgs.Empty)
			End Set
		End Property

		Private Sub _progress_PropertiesChanged(ByVal sender As Object, ByVal e As EventArgs)
			Me.OnPropertiesChanged(EventArgs.Empty)
		End Sub

		Public Sub New()
			MyBase.New()
		End Sub

		Friend Const EditorName As String = "MyTrackBar"

		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyTrackBar), GetType(MyRepositoryItemTrackBar), GetType(MyTrackBarViewInfo), New TrackBarPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property
		Public Overrides Sub Assign(ByVal item As RepositoryItem)
			MyBase.Assign(item)
			Me.Progress = (TryCast(item, MyRepositoryItemTrackBar)).Progress
		End Sub
	End Class

End Namespace
